tools-v2/pkg/cli/command/curvebs/update/leader/leader.go

```diff
@@ -0,0 +1,207 @@
+}
+
+func (lCmd *leaderCommand) Init(cmd *cobra.Command, args []string) error {
+	lCmd.opts = peer.Options{}
+
+	lCmd.SetHeader([]string{cobrautil.ROW_LEADER, cobrautil.ROW_OLDLEADER, cobrautil.ROW_COPYSET, cobrautil.ROW_RESULT, cobrautil.ROW_REASON})
+	lCmd.TableNew.SetAutoMergeCellsByColumnIndex(cobrautil.GetIndexSlice(
+		lCmd.Header, []string{},
+	))
+
+	lCmd.opts.Timeout = config.GetFlagDuration(lCmd.Cmd, config.RPCTIMEOUT)
+	lCmd.opts.RetryTimes = config.GetFlagInt32(lCmd.Cmd, config.RPCRETRYTIMES)
+
+	lCmd.copysetID = config.GetBsFlagUint32(lCmd.Cmd, config.CURVEBS_COPYSET_ID)
+	lCmd.logicalPoolID = config.GetBsFlagUint32(lCmd.Cmd, config.CURVEBS_LOGIC_POOL_ID)
+
+	// parse peers config
+	peers := config.GetBsFlagStringSlice(lCmd.Cmd, config.CURVEBS_PEERS_ADDRESS)
+	c, e := peer.ParseConfiguration(peers)
+	if e != nil {
+		return e.ToError()
+	}
+	lCmd.conf = *c
+
+	// parse peer conf
+	if len(args) < 1 {
+		pErr := cmderror.ErrGetPeer()
+		pErr.Format("should specified the peer address")
+		return pErr.ToError()
+	}
+	peerPara := args[0]
+	lCmd.leaderPeer, e = peer.ParsePeer(peerPara)
+	if e != nil {
+		return e.ToError()
+	}
+	return nil
+}
+
+func (lCmd *leaderCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&lCmd.FinalCurveCmd, lCmd)
+}
+
+func (lCmd *leaderCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	out := make(map[string]string)
+	out[cobrautil.ROW_LEADER] = fmt.Sprintf("%s:%d", lCmd.leaderPeer.GetAddress(), lCmd.leaderPeer.GetId())
+	out[cobrautil.ROW_COPYSET] = fmt.Sprintf("(%d:%d)", lCmd.logicalPoolID, lCmd.copysetID)
+
+	// 1. acquire leader peer info.
+	leader, err := peer.GetLeader(lCmd.logicalPoolID, lCmd.copysetID, lCmd.conf, lCmd.opts)
+	if err != nil {
+		out[cobrautil.ROW_RESULT] = "failed"
+		out[cobrautil.ROW_REASON] = err.ToError().Error()
+		lCmd.Error = err
+		return nil
+	}
+
+	out[cobrautil.ROW_OLDLEADER] = fmt.Sprintf("%s", leader.GetAddress())
+
+	// 2. judge is same?
+	oldLeaderPeer, e := peer.ParsePeer(leader.GetAddress())
+	if e != nil {
+		return e.ToError()
+	}
+
+	if oldLeaderPeer.GetAddress() == lCmd.leaderPeer.GetAddress() {
+		out[cobrautil.ROW_RESULT] = "success"
+		out[cobrautil.ROW_REASON] = "don't need to transfer"
+
+		list := cobrautil.Map2List(out, lCmd.Header)
+		lCmd.TableNew.Append(list)
+		return nil
+	}
+
+	// 3. transfer leader
+	lCmd.Result, err = lCmd.execTransferLeader(oldLeaderPeer)
+	if err != nil {
+		out[cobrautil.ROW_RESULT] = "failed"
+		out[cobrautil.ROW_REASON] = err.ToError().Error()
+		lCmd.Error = err
+		return nil
+	}
+
+	out[cobrautil.ROW_RESULT] = "success"
+	out[cobrautil.ROW_REASON] = "null"
+	list := cobrautil.Map2List(out, lCmd.Header)
+	lCmd.TableNew.Append(list)
+	return nil
+}
+
+func (lCmd *leaderCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&lCmd.FinalCurveCmd)
+}
+
+func (lCmd *leaderCommand) execTransferLeader(oldLeaderPeer *common.Peer) (interface{}, *cmderror.CmdError) {
```

Comment:
Define the translator rpc command like any other command

Reply:
done